### PR TITLE
Add categories to item-list and package location support

### PR DESCRIPTION
**Pull request by allmightyspiff**
#918 

```bash
$ ./slcli order package-locations CLOUD_SERVER
265592   ams01  AMS01 - Amsterdam       AMSTERDAM
814994   ams03  AMS03 - Amsterdam       AMSTERDAM03
1004997  che01  CHE01 - Chennai         CHENNAI
```

```bash
./slcli order item-list 2U_DUAL_E52600_V3_12_DRIVES                                                                                                      av_spyware_protection    MCAFEE_VIRUSSCAN_ANTIVIRUS_WINDOWS                                     McAfee VirusScan Enterprise                                     
bandwidth                BANDWIDTH_0_GB                                                         0 GB Bandwidth                                                  
bandwidth                BANDWIDTH_500_GB                                                       500 GB Bandwidth                                                
bandwidth                BANDWIDTH_UNLIMITED_100_MBPS_UPLINK                                    Unlimited Bandwidth (100 Mbps Uplink)                           
bandwidth                BANDWIDTH_1000_GB                                                      1000 GB Bandwidth                                               
bandwidth                BANDWIDTH_5000_GB                                                      5000 GB Bandwidth                                               
bandwidth                BANDWIDTH_10000_GB                                                     10000 GB Bandwidth                                              
bandwidth                BANDWIDTH_20000_GB                                                     20000 GB Bandwidth                                              
bc_insurance             BUSINESS_CONTINUANCE_INSURANCE                                         Business Continuance Insurance           
```

### CODE REVIEW

**Comment by rlrossiter** [comment]
It looks like the hardware manager and dedicated host manager need to be updated for the get_package_by_key now returning an error if no package exists.

https://github.com/softlayer/softlayer-python/blob/master/SoftLayer/managers/dedicated_host.py#L256-L257

https://github.com/softlayer/softlayer-python/blob/master/SoftLayer/managers/hardware.py#L425-L426

Looks like you already updated the hardware tests to test the new error, dedicated host tests will likely need the same thing.

File: SoftLayer/CLI/order/item_list.py [hunk 1]
```diff
@@ -51,10 +48,23 @@ def cli(env, package_keyname, keyword, category):
...
+            table.add_row([catname, item['keyName'], item['description']])
     env.fout(table)
+
+
+def sort_items(items):
+    """sorts the items into a dictionary of categories, with a list of items"""
+
+    sorted_items = {}
+    for item in items:
+        category = lookup(item, 'itemCategory', 'categoryCode')
```
[the remark this answers is not in the document]
Comment (allmightyspiff) [reply SoftLayer/CLI/order/item_list.py]: I'm not sure I want to include all the other categories as I think that will pollute the output. It should be somewhat obvious that disk0/gpu0 could be used with other diskX/gpuX categories. 
I might try concatenating all the categories into a CSV though, I'll have to see what the output looks like.

**Comment by allmightyspiff** [comment]
```bash
...............................................................................................................................................................................................................................:......................................................:..................................................................................................................................:
:                                                                                                            category                                                                                                           :                       keyName                        :                                                           description                                                            :
:...............................................................................................................................................................................................................................:......................................................:..................................................................................................................................:
:                                                                                                           bandwidth                                                                                                           :                    BANDWIDTH_0_GB                    :                                                          0 GB Bandwidth                                                          :
:                                                                                                           bandwidth                                                                                                           :                   BANDWIDTH_500_GB                   :                                                         500 GB Bandwidth                                                         :
:                                                                                                           bandwidth                                                                                                           :         BANDWIDTH_UNLIMITED_100_MBPS_UPLINK          :                                              Unlimited Bandwidth (100 Mbps Uplink)                                               :
:                                                                                                           bandwidth                                                                                                           :                  BANDWIDTH_1000_GB                   :                                                        1000 GB Bandwidth                                                         :
:                                                                                                           bandwidth                                                                                                           :                  BANDWIDTH_5000_GB                   :                                                        5000 GB Bandwidth                                                         :
:                                                                                                           bandwidth                                                                                                           :                  BANDWIDTH_10000_GB                  :                                                        10000 GB Bandwidth                                                        :
:                                                                                                           bandwidth                                                                                                           :                   BANDWIDTH_0_GB_2                   :                                                          0 GB Bandwidth                                                          :
:                                                                                                           bandwidth                                                                                                           :                  BANDWIDTH_20000_GB                  :                                                        20000 GB Bandwidth                                                        :
:         disk0disk9disk14disk21disk33disk6disk11disk22disk27disk29disk12disk13disk15disk17disk18disk23disk31disk2disk24disk3disk16disk20disk4disk7disk19disk28disk32disk34disk5disk8disk10disk25disk35disk1disk26disk30        :                 HARD_DRIVE_800GB_SSD                 :                                                       800 GB SSD (10 DWPD)                                                       :
```

I'll just add something about disk and gpu categories to the help message for this function. Adding them to the output doesn't seem to be worth the extra effort.

File: SoftLayer/CLI/order/item_list.py [hunk 2]
```diff
@@ -23,13 +23,20 @@ def cli(env, package_keyname, keyword, category):
 
     Package keynames can be retrieved using `slcli order package-list`
 
+    \b
+    Note:
+        Items with a numbered category, like disk0 or gpu0, need to be included
+        multiple times in an order to match how many of the item you want to order.
```
Comment (rlrossiter) [inline SoftLayer/CLI/order/item_list.py]: Nitpicky, but I'd like this to read something more like this:

> Items with a numbered category, like disk0 or gpu0, can be included multiple times in an order to attach more than one of that item to the order.

The way it reads right now is it is _required_ to specify numbered-category items more than once on an order.

**Pull request merged**